src/mad_spark_alt/evolution/genetic_algorithm.py

```
@@ -67,6 +72,7 @@ def __init__(
         cache_ttl: int = 3600,
         checkpoint_dir: Optional[str] = None,
         checkpoint_interval: int = 0,
+        llm_provider = None,
```

Ruff (0.12.2)

Missing type annotation for function argument `llm_provider`

(ANN001)